### PR TITLE
WIP: Remove deprecated 2.9 network settings

### PR DESCRIPTION
**Pull request by pabelanger**
##### SUMMARY
Remove deprecated settings from network modules

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ansible network modules

### CODE REVIEW

File: lib/ansible/module_utils/network/asa/asa.py [hunk 1]
```diff
@@ -52,14 +52,7 @@
 }
 
 asa_top_spec = {
-    'host': dict(removed_in_version=2.9),
-    'port': dict(removed_in_version=2.9, type='int'),
-    'username': dict(removed_in_version=2.9),
-    'password': dict(removed_in_version=2.9, no_log=True),
-    'ssh_keyfile': dict(removed_in_version=2.9, type='path'),
     'authorize': dict(type='bool'),
```
Comment (pabelanger) [inline lib/ansible/module_utils/network/asa/asa.py]: So, this wasn't marked to be removed, but I think we want to.

File: lib/ansible/module_utils/network/eos/eos.py [hunk 2]
```diff
@@ -62,20 +62,7 @@
     'provider': dict(type='dict', options=eos_provider_spec),
 }
 eos_top_spec = {
-    'host': dict(removed_in_version=2.9),
-    'port': dict(removed_in_version=2.9, type='int'),
-    'username': dict(removed_in_version=2.9),
-    'password': dict(removed_in_version=2.9, no_log=True),
-    'ssh_keyfile': dict(removed_in_version=2.9, type='path'),
-
     'authorize': dict(fallback=(env_fallback, ['ANSIBLE_NET_AUTHORIZE']), type='bool'),
```
Comment (pabelanger) [inline lib/ansible/module_utils/network/eos/eos.py]: same comment

File: lib/ansible/module_utils/network/ios/ios.py [hunk 3]
```diff
@@ -49,14 +49,7 @@
 }
 
 ios_top_spec = {
-    'host': dict(removed_in_version=2.9),
-    'port': dict(removed_in_version=2.9, type='int'),
-    'username': dict(removed_in_version=2.9),
-    'password': dict(removed_in_version=2.9, no_log=True),
-    'ssh_keyfile': dict(removed_in_version=2.9, type='path'),
     'authorize': dict(fallback=(env_fallback, ['ANSIBLE_NET_AUTHORIZE']), type='bool'),
```
Comment (pabelanger) [inline lib/ansible/module_utils/network/ios/ios.py]: same

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 4]
```diff
@@ -82,21 +82,7 @@
...
 }
 nxos_top_spec = {
-    'host': dict(type='str', removed_in_version=2.9),
-    'port': dict(type='int', removed_in_version=2.9),
-
-    'username': dict(type='str', removed_in_version=2.9),
-    'password': dict(type='str', no_log=True, removed_in_version=2.9),
-    'ssh_keyfile': dict(type='str', removed_in_version=2.9),
-
     'authorize': dict(type='bool', fallback=(env_fallback, ['ANSIBLE_NET_AUTHORIZE'])),
```
Comment (pabelanger) [inline lib/ansible/module_utils/network/nxos/nxos.py]: same

File: lib/ansible/module_utils/network/asa/asa.py (hunk 1 above)
Comment (ganeshrn) [reply lib/ansible/module_utils/network/asa/asa.py]: It seems this option was missed out. We can remove it since `authorize` is also part of `provider`

**Comment by pabelanger** [comment]
Pull-request updated, HEAD is now 8d57517eafd10d2f7c4f6791a1fe0b7ca8a5b553

**Comment by pabelanger** [comment]
Pull-request updated, HEAD is now da4dd36d0e89ae97f381feab0fb949722900cc05

**Review by Qalthos — CHANGES_REQUESTED** [review]
You should also remove the top-level arguments from the platforms' doc fragments in plugins/doc_fragments, where necessary.

**Comment by pabelanger** [comment]
recheck

**Pull request closed without merging**